File: middle-of-the-linked-list/middle-of-the-linked-list.cpp
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        int count = 0;
        ListNode* dummy = head;
        while (dummy) {
            count++;
            dummy = dummy -> next;
        }
        count /= 2;
        ListNode* cur = head;
        while (cur && count) {
            count--;
            cur = cur -> next;
        }
        return cur;
    }
};

File: middle-of-the-linked-list/middle-of-the-linked-list.py
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        count = 0
        dummy = head
        while dummy:
            count += 1
            dummy = dummy.next
        cur = head
        count //= 2
        while cur and count:
            count -= 1
            cur = cur.next
        return cur
